yarn-project/simulator/src/client/view_data_oracle.ts

```diff
@@ -364,10 +364,12 @@ export class ViewDataOracle extends TypedOracle {
 
   // TODO(#11849): consider replacing this oracle with a pure Noir implementation of aes decryption.
   public override aes128Decrypt(ciphertext: Buffer, iv: Buffer, symKey: Buffer): Promise<Buffer> {
-    // Noir can't predict the amount of padding that gets trimmed,
-    // but it needs to know the length of the returned value.
-    // So we tell Noir that the length is the (predictable) length
-    // of the padded plaintext, we return that padded plaintext, and have Noir interpret the padding to do the trimming.
+    // We do not trim the padding in typescript-land.
+    // In Noir-land, when this oracle is called, all the caller knows is the
+    // ciphertext length. Without performing decryption, the caller cannot know the
+    // true plaintext length. Therefore, the array length of the return type of this
+    // oracle must be that of the ciphertext, which is equal to the _padded_ plaintext.
+    // It's then up to the Noir code to remove the pkcs#7 padding itself.
```

Comment:
Not a fan of manually having to remove the padding tbh. I'd rather return a BoundedVec. We can of course do that in Noir.